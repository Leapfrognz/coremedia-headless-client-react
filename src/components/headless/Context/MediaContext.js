// @flow
import React from 'react';

const MediaContext = React.createContext({
  getMediaUrl: (link: string): string => link,
});

export default MediaContext;
