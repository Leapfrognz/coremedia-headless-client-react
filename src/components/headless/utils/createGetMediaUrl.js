// @flow
const createGetMediaUrl = (baseURL: string, prefix: string) => (
  link: string,
  ratio?: string,
  minWidth?: number
): string => {
  if (link && link.startsWith(prefix)) {
    const parts = link.substring(prefix.length).split('/');
    if (parts.length === 3) {
      const [type, id, property] = parts;
      switch (type) {
        case 'image': {
          const mediaUrl = `${baseURL}/media/${id}/${property}`;
          let queryParams = '';
          if (ratio) {
            queryParams = `?ratio=${ratio}`;
          }
          if (minWidth) {
            queryParams = `${!queryParams ? '?' : `${queryParams}&`}minWidth=${minWidth}`;
          }
          return mediaUrl + queryParams;
        }
        case 'media': {
          return baseURL + `/media/${id}/${property}`;
        }
        default:
          return '';
      }
    }
  }
  return '';
};

export default createGetMediaUrl;
