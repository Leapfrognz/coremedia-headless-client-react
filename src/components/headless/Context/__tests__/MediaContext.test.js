// @flow
import React from 'react';
import renderer from 'react-test-renderer';

import MediaContext from '../MediaContext';
import createGetMediaUrl from '../../utils/createGetMediaUrl';

describe('MediaContext', () => {
  it('should render correctly with default value', () => {
    const tree = renderer
      .create(
        <MediaContext.Consumer>
          {value => value.getMediaUrl('http://localhost/image.jpg')}
        </MediaContext.Consumer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly with provided value', () => {
    const getMediaUrl = createGetMediaUrl('http://localhost', 'coremedia:///');
    const tree = renderer
      .create(
        <MediaContext.Provider value={{ getMediaUrl }}>
          <MediaContext.Consumer>
            {value => value.getMediaUrl('coremedia:///image/7564/data')}
          </MediaContext.Consumer>
        </MediaContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
