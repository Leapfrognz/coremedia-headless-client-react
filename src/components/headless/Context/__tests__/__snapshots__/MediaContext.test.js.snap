// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaContext should render correctly with default value 1`] = `"http://localhost/image.jpg"`;

exports[`MediaContext should render correctly with provided value 1`] = `"http://localhost/media/7564/data"`;
